pkgs/by-name/so/sobjectizer/package.nix

```diff
@@ -0,0 +1,51 @@
+{
+  pkgs,
+  lib,
+  stdenv,
+  fetchFromGitHub,
+  cmake,
+  withStatic ? true,
+  withShared ? true,
+  buildExamples ? false,
+  buildTests ? true,
+  buildAll ? false,
+}:
+
+stdenv.mkDerivation (finalAttrs: {
+  pname = "sobjectizer";
+  version = "5.8.4";
+
+  src = fetchFromGitHub {
+    owner = "Stiffstream";
+    repo = "sobjectizer";
+    tag = "v.${finalAttrs.version}";
+    hash = "sha256-tIqWgd6TppHfqZk3XHzhG0t+Nn8BQCTP81UD7ls67UE=";
+  };
+
+  nativeBuildInputs = [ cmake ];
+
+  cmakeDir = "../dev";
+
+  cmakeFlags = [
+    (lib.cmakeBool "SOBJECTIZER_BUILD_STATIC" withStatic)
+    (lib.cmakeBool "SOBJECTIZER_BUILD_SHARED" withShared)
+    (lib.cmakeBool "BUILD_EXAMPLES" buildExamples)
+    (lib.cmakeBool "BUILD_TESTS" buildTests)
+    (lib.cmakeBool "BUILD_ALL" buildAll)
+  ];
```

Comment:
```nix
cmakeFlags = [
  (lib.cmakeBool "SOBJECTIZER_BUILD_STATIC" withStatic)
  (lib.cmakeBool "SOBJECTIZER_BUILD_SHARED" withShared)
  (lib.cmakeBool "BUILD_EXAMPLES" (buildExamples && withStatic))
  (lib.cmakeBool "BUILD_TESTS" (finalAttrs.doCheck && withShared))
];
```

The examples require the static library, but, when applying the patch, the tests do not.

Reply:
To add on to this: 
`(lib.cmakeBool "BUILD_EXAMPLES" (buildExamples && withStatic))` 
can be reduced to
`lib.cmakeBool "BUILD_EXAMPLES" buildExamples)`
if the package also includes `assert buildExamples -> withStatic;`